### PR TITLE
[WIP] Integration tests with real docker

### PR DESCRIPTION
**Pull request by iRevive**
This issue covers https://github.com/gvolpe/fs2-rabbit/issues/126.

Hi @gvolpe. I have implemented several integration tests with a real docker container.
Currently, it covers only two main scenarios: 
1) Connection management;  
2) Publishing/consuming messages from the queue.

Do you have any ideas about additional tests?

### CODE REVIEW

File: core/src/test/scala/com/github/gvolpe/fs2rabbit/DockerRabbit.scala [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+import org.scalatest.{BeforeAndAfterAll, Suite}
+
+import scala.concurrent.{ExecutionContext, SyncVar}
+import scala.sys.process._
+
+trait DockerRabbit extends BeforeAndAfterAll { self: Suite =>
+
+  import DockerRabbit._
+
+  // override this if the Redis container has to be started before invocation
```
Comment (gvolpe) [inline core/src/test/scala/com/github/gvolpe/fs2rabbit/DockerRabbit.scala]: Replace `Redis` by `RabbitMQ` :wink:

File: core/src/test/scala/com/github/gvolpe/fs2rabbit/IOAssertion.scala [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ */
+
+package com.github.gvolpe.fs2rabbit
+
+import cats.effect.IO
+import cats.syntax.functor._
+
+object IOAssertion {
+
+  def apply[A](flow: IO[A]): Unit = flow.void.unsafeRunSync()
```
Comment (gvolpe) [inline core/src/test/scala/com/github/gvolpe/fs2rabbit/IOAssertion.scala]: I'm curious, why `flow`? I normally name this `ioa` for `IO[A]` :)

File: core/src/test/scala/com/github/gvolpe/fs2rabbit/DockerRabbit.scala (hunk 1 above)
Comment (iRevive) [reply core/src/test/scala/com/github/gvolpe/fs2rabbit/DockerRabbit.scala]: Good catch, thanks.

**Comment by iRevive** [comment]
I will add additional test soon. Independent subproject with tests sounds like a good idea.

File: core/src/test/scala/com/github/gvolpe/fs2rabbit/IOAssertion.scala (hunk 2 above)
Comment (iRevive) [reply core/src/test/scala/com/github/gvolpe/fs2rabbit/IOAssertion.scala]: Done

**Comment by iRevive** [comment]
I moved tests into a new subproject. Also, I removed existing tests with mocked RabbitMQ. 
Can you take a look @gvolpe?

**Pull request merged**